src/components/composer/composer/cell_composer_store.ts

```diff
@@ -0,0 +1,298 @@
+      this.model.dispatch("UPDATE_CELL", {
+        ...this.currentEditedCell,
+        content: "",
+      });
+    }
+    this.model.dispatch("AUTOFILL_TABLE_COLUMN", { ...this.currentEditedCell });
+    this.setContent("");
+  }
+
+  protected getComposerContent(position: CellPosition): string {
+    const locale = this.getters.getLocale();
+    const cell = this.getters.getCell(position);
+    if (cell?.isFormula) {
+      return localizeFormula(cell.content, locale);
+    }
+    const { format, value, type, formattedValue } = this.getters.getEvaluatedCell(position);
+    switch (type) {
+      case CellValueType.empty:
+        return "";
+      case CellValueType.text:
+      case CellValueType.error:
+        return value;
+      case CellValueType.boolean:
+        return formattedValue;
+      case CellValueType.number:
+        if (format && isDateTimeFormat(format)) {
+          if (parseDateTime(formattedValue, locale) !== null) {
+            // formatted string can be parsed again
+            return formattedValue;
+          }
+          // display a simplified and parsable string otherwise
+          const timeFormat = Number.isInteger(value)
+            ? locale.dateFormat
+            : getDateTimeFormat(locale);
+          return formatValue(value, { locale, format: timeFormat });
+        }
+        return this.numberComposerContent(value, format, locale);
+    }
+  }
+
+  private numberComposerContent(value: number, format: Format | undefined, locale: Locale): string {
+    if (format?.includes("%")) {
+      return `${numberToString(value * 100, locale.decimalSeparator)}%`;
+    }
+    return numberToString(value, locale.decimalSeparator);
+  }
+
+  /** Add headers at the end of the sheet so the formula in the composer has enough space to spread */
+  private addHeadersForSpreadingFormula(content: string) {
+    if (!content.startsWith("=")) {
+      return;
+    }
+
+    const evaluated = this.getters.evaluateFormula(this.sheetId, content);
+    if (!isMatrix(evaluated)) {
+      return;
+    }
+
+    const numberOfRows = this.getters.getNumberRows(this.sheetId);
+    const numberOfCols = this.getters.getNumberCols(this.sheetId);
+
+    const missingRows = this.row + evaluated[0].length - numberOfRows;
+    const missingCols = this.col + evaluated.length - numberOfCols;
+
+    if (missingCols > 0) {
+      this.model.dispatch("ADD_COLUMNS_ROWS", {
+        sheetId: this.sheetId,
+        dimension: "COL",
+        base: numberOfCols - 1,
+        position: "after",
+        quantity: missingCols + 20,
+      });
+    }
+    if (missingRows > 0) {
+      this.model.dispatch("ADD_COLUMNS_ROWS", {
+        sheetId: this.sheetId,
+        dimension: "ROW",
+        base: numberOfRows - 1,
+        position: "after",
+        quantity: missingRows + 50,
+      });
+    }
+  }
+
+  private checkDataValidation(): boolean {
+    const cellPosition = { sheetId: this.sheetId, col: this.col, row: this.row };
+    try {
```

Comment:
for later: I think we can remove the try/catch. I think it was for `evaluateFormula` which doesn't throw anymore

Reply:
added a commit for it